testsuite/shared/src/main/java/org/jboss/as/test/shared/LayersTestBase.java

```diff
@@ -179,16 +205,115 @@ public class LayersTestBase {
+            // test-all-layers installation is non-ha and does not include layers that provide jgroups
+            "org.jboss.as.clustering.jgroups",
+            // TODO we need to add an agroal layer
+            "org.wildfly.extension.datasources-agroal",
+            "io.agroal",
+            // Legacy subsystems for which we will not provide layers.
+            // Not in the ootb standalone.xml extension list
+            "org.wildfly.extension.picketlink",
+            "org.jboss.as.jsr77",
+            "org.keycloak.keycloak-adapter-subsystem",
+            // end legacy subsystems ^^^
+            // TODO move eclipse link support to an external feature pack
+            "org.eclipse.persistence",
+            // RA not associated with any layer
+            "org.jboss.genericjms",
+            // Appclient support is not provided by a layer
+            "org.jboss.as.appclient",
+            "org.jboss.metadata.appclient",
+            // TODO WFLY-16576 -- cruft?
+            "org.bouncycastle",
+            // This was brought in as part an RFE, WFLY-10632 & WFLY-10636. While the module is currently marked as private,
+            // for now we should keep this module.
+            "org.jboss.resteasy.resteasy-rxjava2",
+            // TODO these implement SPIs from RESTEasy or JBoss WS but I don't know how they integrate
+            // as there is no ref to them in any module.xml nor any in WF java code.
+            // Perhaps via deployment descriptor? In any case, no layer provides them
+            "org.wildfly.security.jakarta.client.resteasy",
+            "org.wildfly.security.jakarta.client.webservices",
+            // TODO we need to add an rts layer
+            "org.wildfly.extension.rts",
+            "org.jboss.narayana.rts",
+    };
+
+    /**
+     * Included in the return value of both {@link #getExpectedUnusedInAllLayers()}
+     * and {@link #getExpectedUnreferenced()}, but only when testing provisioning
+     * directly from the wildfly-ee feature pack.
+     */
+    public static final String[] NO_LAYER_OR_REFERENCE_WILDFLY_EE = {
+            // In wildfly-ee only referenced by the
+            // unused-in-all-layers org.jboss.resteasy.resteasy-rxjava2
+            "io.reactivex.rxjava2.rxjava"
+    };
+
+    /**
+     * Utility method to combine various module name arrays into sets for use
+     * as parameters to this class' methods.
+     *
+     * @param first the first array to combine. Cannot be {@code null}
+     * @param others other arrays to combine. Can be {@code null}
+     * @return a set containing all of the elements in the arrays
+     */
+    @SuppressWarnings("SameParameterValue")
+    protected static Set<String> concatArrays(String[] first, String[]... others) {
+        // TODO Delegate to LayersTest.concatArrays once WFCORE-6481 is integrated
+        if (others == null || others.length == 0) {
+            return new HashSet<>(Arrays.asList(first));
```

Comment:
Really not important, but we could use `Set.of(first)`.

Reply:
Good point; old habits die hard. I'll do that in the wf core code that will replace this.